### ISSUE TITLE
Reliable scheduler mangles large numeric arguments

### ISSUE THREAD

**Issue by mperham** [mperham/sidekiq#2478]
Large numeric arguments do not survive the JSON roundtrip within Lua, needed by the reliable scheduler.  Reproduced:

```ruby
    result = Sidekiq.redis do |conn|
      conn.eval <<-LUA
        local text = '[130775545883395173]'
        local job = cjson.decode(text)
        return cjson.encode(job)
      LUA
    end
    p result
```

Not sure if this is fixable or not.

**Comment by mperham** [comment]
Ruby's JSON works perfectly:

```
   p JSON.dump(JSON.load('[130775545883395173]'))
```

**Comment by aprescott** [comment]
I suspect this won't be fixable with cjson. Its manual specifies that [`cjson.encode` uses a precision of 14 significant digits](http://www.kyne.com.au/~mark/software/lua-cjson-manual.html#encode), out of a [maximum precision of 14](http://www.kyne.com.au/~mark/software/lua-cjson-manual.html#encode_number_precision). (Although my output from your `result` code has 13 digits, even when adding more digits.)

Seems like someone tried to work around this in https://github.com/mpx/lua-cjson/pull/9.

**Comment by mperham** [comment]
It looks like the CJSON library used by Lua only supports numbers up to 14 digits of precision.  This is terrible, it can't encode 64 bit integers correctly.

http://www.kyne.com.au/~mark/software/lua-cjson-manual.html#encode_number_precision

http://stackoverflow.com/questions/24886369/send-a-large-number-from-php-to-jquery-without-scientific-notation

**Comment by aprescott** [comment]
Haha, snap!

**Comment by mperham** [comment]
Not sure how long I'll need to shower after releasing this hack:

```lua
string.gsub(text, '(\"enqueued_at\":)[0-9]+', '%1'..now)
```

**Comment by aprescott** [comment]
:scream:

**Comment by mperham** [comment]
Released 2.0.8.
